alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt

```diff
@@ -122,7 +125,9 @@ class AlertService(
     fun composeQueryLevelAlert(
         ctx: QueryLevelTriggerExecutionContext,
         result: QueryLevelTriggerRunResult,
-        alertError: AlertError?
+        alertError: AlertError?,
+        executionId: String? = null,
```

Comment:
Did we want the executionId to be optional? I remember conversations about this being required to help with auditing in the future.

Reply:
fixed.

made it mandatory in monitors.